### PR TITLE
feat: add export ErrorObject

### PR DESCRIPTION
**Pull request by budarin**
Export ErrorObject to support fastify PR [#4283](https://github.com/fastify/fastify/pull/4283)

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [~] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Thanks for opening a PR! Can you please add a unit test? We use tsd for typescript.

File: index.d.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export type ValidatorCompiler = (
 
 export declare function ValidatorSelector(): ValidatorCompiler;
 
-export type { Options } from "ajv";
+export type { Options, ErrorObject } from "ajv";
```
Comment (Eomm) [inline index.d.ts:32]: Could you add a test to this file https://github.com/fastify/ajv-compiler/blob/main/test/types/index.test.ts?

Comment (budarin) [reply index.d.ts:32]: I can't figure out how to make a type test that is exported from the same module

Comment (Eomm) [reply index.d.ts:32]: It should be doable by importing the same from both the modules and `expectType<ErrorFromThisModule>(ErrorImportedFromAjv)`

By doing this we assure that we don't remove the Error exportorted types

Comment (Uzlopak) [reply index.d.ts:32]:
```typescript
expectType<unknown>(({} as ErrorObject).data)
expectType<string>(({} as ErrorObject).instancePath)
expectType<string>(({} as ErrorObject).keyword)
expectType<string | undefined>(({} as ErrorObject).message)
expectType<Record<string, any>>(({} as ErrorObject).params)
expectType<AnySchemaObject | undefined>(({} as ErrorObject).parentSchema)
expectType<string | undefined>(({} as ErrorObject).propertyName)
expectType<unknown>(({} as ErrorObject).schema)
expectType<string>(({} as ErrorObject).schemaPath)
```

Comment (budarin) [reply index.d.ts:32]: but this module must have import the type from `ajv` or construct in this module (which is not possible) and compare the type from 'ajv' and exported

Sorry, I don't understand how to write it...

Comment (budarin) [reply index.d.ts:32]: > ```ts
> expectType<unknown>(({} as ErrorObject).data)
> expectType<string>(({} as ErrorObject).instancePath)
> expectType<string>(({} as ErrorObject).keyword)
> expectType<string | undefined>(({} as ErrorObject).message)
> expectType<Record<string, any>>(({} as ErrorObject).params)
> expectType<AnySchemaObject | undefined>(({} as ErrorObject).parentSchema)
> expectType<string | undefined>(({} as ErrorObject).propertyName)
> expectType<unknown>(({} as ErrorObject).schema)
> expectType<string>(({} as ErrorObject).schemaPath)
> ```

can I take this code?

Comment (Uzlopak) [reply index.d.ts:32]: Sure.

**Review by Eomm — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**